account_multicompany_easy_creation/wizards/multicompany_easy_creation.py

```diff
@@ -157,8 +157,19 @@ def create_company(self):
         self.new_company_id = self.env["res.company"].create(
             {"name": self.name, "user_ids": [(6, 0, self.user_ids.ids)]}
         )
-        self.new_company_id.chart_template_id = self.chart_template_id
-        self.new_company_id.chart_template_id.try_loading_for_current_company()
+        self.env.user.write(
+            {
+                "company_ids": [(4, self.new_company_id.id)],
+                "company_id": self.new_company_id.id,
+            }
+        )
+        ctx = dict(self.env.context)
+        allowed_company_ids = ctx.get("allowed_company_ids") or []
+        allowed_company_ids.append(self.new_company_id.id)
+        ctx.update(allowed_company_ids=allowed_company_ids)
```

Comment:
Is this really needed?

Reply:
After many tests in local I have discovered that this is necessary to avoid reading errors in all the models in which you try to create records by the access rules.